src/main/scala/com/datastax/gatling/plugin/utils/ResultSetUtils.scala

```diff
@@ -0,0 +1,77 @@
+package com.datastax.gatling.plugin.utils
+
+import java.util.concurrent.CompletionStage
+
+import com.datastax.dse.driver.api.core.graph.{AsyncGraphResultSet, GraphNode}
+import com.datastax.oss.driver.api.core.cql.{AsyncResultSet, Row}
+import org.apache.tinkerpop.gremlin.process.traversal.Path
+import org.apache.tinkerpop.gremlin.structure.{Edge, Property, Vertex, VertexProperty}
+
+import collection.JavaConverters._
+
+object ResultSetUtils {
+
+  def asyncResultSetToIterator(resultSet:AsyncResultSet): Iterator[Row] =
+    new AsyncResultSetIterator(
+      new CqlAsyncRS(resultSet))
+
+  def asyncGraphResultSetToIterator(resultSet:AsyncGraphResultSet): Iterator[GraphNode] =
+    new AsyncResultSetIterator(
+      new GraphAsyncRS(resultSet))
+}
+
+object GraphResultSetUtils {
+
+  private def buildFilterAndMapFn[T]
+  (filterFn: GraphNode => Boolean, mapFn: GraphNode => T)
+  (resultSet: AsyncGraphResultSet, key:String):Iterator[T] = {
+    ResultSetUtils.asyncGraphResultSetToIterator(resultSet )
+      .map(graphNode => graphNode.getByKey(key))
+      .filter(filterFn)
+      .map(mapFn)
+  }
+
+  def edges: (AsyncGraphResultSet, String) => Iterator[Edge] = buildFilterAndMapFn(_.isEdge, _.asEdge)
+
+  def vertexes: (AsyncGraphResultSet, String) => Iterator[Vertex] = buildFilterAndMapFn(_.isVertex, _.asVertex)
+
+  def paths: (AsyncGraphResultSet, String) => Iterator[Path] = buildFilterAndMapFn(_.isPath, _.asPath)
+
+  def properties: (AsyncGraphResultSet, String) => Iterator[Property[_]] =
+    buildFilterAndMapFn(_.isProperty, _.asProperty.asInstanceOf[Property[_]])
+
+  def vertexProperties: (AsyncGraphResultSet, String) => Iterator[VertexProperty[_]] =
+    buildFilterAndMapFn(_.isVertexProperty, _.asVertexProperty.asInstanceOf[VertexProperty[_]])
+}
+
+trait AsyncRS[T] {
+  def hasMorePages:Boolean
```

Comment:
I think that you didn't run formatter plugin

Reply:
Probably true, although I think the def for AsyncRS stays pretty much the same.  I'll push the formatted file up soon though.